### PR TITLE
Restrict repository to run daily.yml

### PR DESCRIPTION
**Pull request by YaSuenag**
This PR is for #1326.

I added `github.repository == 'sustainable-computing-io/kepler'` to prevent to run daily.yml on forked repo. Similar code works fine on [Carbon Aware SDK by Green Software Foundation](https://github.com/Green-Software-Foundation/carbon-aware-sdk).

https://github.com/Green-Software-Foundation/carbon-aware-sdk/commit/06c5577397815f2e867a349da7c74be06a9423d1

### CODE REVIEW

**Comment by YaSuenag** [comment]
I saw similar error in push.yml, so I also fixed it.

File: .github/workflows/push.yml
Comment (SamYuan1990) [inline .github/workflows/push.yml:14]: if so.... 
https://github.com/sustainable-computing-io/kepler/pull/1327/files#diff-f3fc934cf0d89bdf07de358896ff90f1202585df812cf606206d1830a9949811R20
https://github.com/sustainable-computing-io/kepler/pull/1327/files#diff-f3fc934cf0d89bdf07de358896ff90f1202585df812cf606206d1830a9949811R25
https://github.com/sustainable-computing-io/kepler/pull/1327/files#diff-f3fc934cf0d89bdf07de358896ff90f1202585df812cf606206d1830a9949811R25
also need for this check.

Comment (YaSuenag) [reply .github/workflows/push.yml:14]: I can add same condition to them, but they succeeded in this PR at least. That may be because it does not include code changes, but I'm not sure.

So should I add?

**Review by SamYuan1990 — COMMENTED** [review]
btw, I am interesting with the job if which not failure.
Does it mean the job will run success as an empty for each fork repo?
for my own fork, I disabled manually to avoid meaningless fork daily build.

File: .github/workflows/push.yml
Comment (SamYuan1990) [reply .github/workflows/push.yml:14]: from the condition, which means when you merge a PR on main branch and meet the specific path then the job been run as https://github.com/sustainable-computing-io/kepler/pull/1327/files#diff-f3fc934cf0d89bdf07de358896ff90f1202585df812cf606206d1830a9949811R4-R9
so, in fact, the changes on this file .... considering with running at a fork repo. only been triggered when the fork owner merge a PR at the fork side.

Comment (SamYuan1990) [reply .github/workflows/push.yml:14]: for example, as https://github.com/SamYuan1990/kepler/pull/8 this is a PR at fork side, and once the pr merged. this pipeline been triggered..... if we only focus on daily build, we should not change on this file.

**Review by SamYuan1990 — CHANGES_REQUESTED** [review]
some discussion on going and tested may needed.
mark as request changes and don't merge before we clean up with discussion.

**Comment by YaSuenag** [comment]
> Does it mean the job will run success as an empty for each fork repo?

It would be skipped.

File: .github/workflows/push.yml
Comment (YaSuenag) [reply .github/workflows/push.yml:14]: > for example, as [SamYuan1990#8](https://github.com/SamYuan1990/kepler/pull/8) this is a PR at fork side, and once the pr merged. this pipeline been triggered..... if we only focus on daily build, we should not change on this file.

You are right. I added this change because I just found it. So I can remove if it needs.

**Comment by SamYuan1990** [comment]
> > Does it mean the job will run success as an empty for each fork repo?
> 
> It would be skipped.

@YaSuenag , it would be skipped as the job will run but as each pipeline doesn't meet the condition, hence we will see a job with everything skipped? or we should better disable the job at meanwhile?

as we have 153 fork repos today, this pr will safe the 153 fork from running workloads everyday.
and we should also suggestion 153 forks also disable the job to avoid an empty running?

**Comment by SamYuan1990** [comment]
would you mind to add a reminder at https://github.com/sustainable-computing-io/kepler/blob/main/CONTRIBUTING.md to disable the job to avoid empty skip? or I can open another PR for that.

File: .github/workflows/push.yml
Comment (SamYuan1990) [reply .github/workflows/push.yml:14]: IMO, I recommend PR back to upstream, but not a PR just on a specific fork.
do we have any specific case to merge a PR at local only?

Comment (SamYuan1990) [reply .github/workflows/push.yml:14]: ok, so let's just have daily.yml in this PR. :-)

Comment (YaSuenag) [reply .github/workflows/push.yml]: I reverted change in push.yml.

**Comment by YaSuenag** [comment]
I can squash, but I have to do force push. Is it ok?
IMHO it is better to do squash merge when this PR can go.

**Comment by YaSuenag** [comment]
@SamYuan1990 

> could you please try git squash?

Done!

**Review by SamYuan1990 — APPROVED** [review]
LGTM

**Pull request merged**
